testcases/open_posix_testsuite/conformance/interfaces/open/1-1.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is licensed under the GPL license.  For the full content
+ * of this license, see the COPYING file at the top level of this
+ * source tree.
+ */
+
+/*
+ * assertion:
+ * Establish the connection between file and file descriptor
+ *
+ * method:
+ *	-open file in write mode using open()
+ *	-write 1 sample string into a file
+ *	-close the file
+ *	-open the same file in read mode
+ *	-read data from the file
+ *	-compare the value read with the value written in first step
+ */
+
+#define _XOPEN_SOURCE 700
+#include <stdio.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include "posixtest.h"
+
+#define TNAME "open/1-1.c"
+#define FNAME "./test_file"
+#define STR_CONST "POSIX CONFORMANCE TEST"
+
+static void remove_file(int fd)
+{
+	errno = 0;
+	if (fd > 0) {
+		if (close(fd) < 0) {
+			printf(TNAME " Error at close(), errno = %d, %s \n", errno, strerror(errno));
+			unlink(FNAME);
+			exit(PTS_UNRESOLVED);
+		}
+	}
+	unlink(FNAME);
+}
+
+int main(void)
+{
+	int test_fd, ret;
+	char read_value[sizeof(STR_CONST)] = {};
+	errno = 0;
+
+	test_fd = open(FNAME, O_CREAT | O_WRONLY, S_IRWXG | S_IRWXU);
+	if (test_fd < 0) {
+		printf(TNAME " Error in open(), errno: %d, %s\n", errno, strerror(errno));
+		exit(PTS_UNRESOLVED);
+	}
+
+	ret = write(test_fd, STR_CONST, sizeof(STR_CONST));
+	if (ret < 0) {
+		printf(TNAME " Error in write()\n");
+		remove_file(test_fd);
+		exit(PTS_UNRESOLVED);
+	}
+	errno = 0;
+	if (close(test_fd) < 0) {
+		printf(TNAME " Error at close(), errno = %d, %s\n", errno, strerror(errno));
+		unlink(FNAME);
+		exit(PTS_UNRESOLVED);
+	}
+
+	errno = 0;
+	test_fd = open(FNAME, O_RDONLY);
+	if (test_fd < 0) {
+		printf(TNAME " Error in open(), errno = %d, %s\n", errno, strerror(errno));
+		remove_file(test_fd);
+		exit(PTS_UNRESOLVED);
+	}
+
+	errno = 0;
+	ret = read(test_fd, read_value, sizeof(STR_CONST));
```

Comment:
Here as well read can return less than size bytes. It will likely not happen here as we read just a few bytes from a file but you should keep that in mind and at least check if we happened to write the whole string or not.

Reply:
updated the logic to check for only bytes that were written.